src/main/scala/workflow/DelegatingTransformer.scala

```diff
@@ -0,0 +1,13 @@
+package workflow
+
+import org.apache.spark.rdd.RDD
+
+/**
+ * A transformer used internally to apply the fit output of an Estimator onto data dependencies.
+ * Takes one fit dependency, and directly applies its transform onto the data dependencies.
+ */
+private[workflow] class DelegatingTransformer[T](override val label: String) extends TransformerNode[T] {
+  def transform(dataDependencies: Seq[_], fitDependencies: Seq[TransformerNode[_]]): T = fitDependencies.head.transform(dataDependencies, Seq()).asInstanceOf[T]
```

Comment:
Could we document above that we use the first fitDependency as that seems kind of important ? Also some reasoning as to why that is okay would be good to have